src/Package/Impl/Shell/VsAppShell.cs

```diff
@@ -47,9 +50,29 @@ namespace Microsoft.VisualStudio.R.Package.Shell {
     public sealed class VsAppShell : IApplicationShell, IMainThread, IIdleTimeService, IDisposable {
         private static VsAppShell _instance;
         private static IApplicationShell _testShell;
+
+        private readonly ITelemetryService _telemetryService;
+        private readonly IRSettings _settings;
+        private readonly ICoreServices _coreServices;
+        private readonly IApplicationConstants _appConstants;
+
         private IdleTimeSource _idleTimeSource;
         private IWritableSettingsStorage _settingStorage;
 
+        [ImportingConstructor]
+        public VsAppShell(ITelemetryService telemetryService
+            , IRSettings settings
+            , ICoreServices coreServices
+            , IApplicationConstants appConstants) {
+
+            _telemetryService = telemetryService;
+            _coreServices = coreServices;
+            _appConstants = appConstants;
+
+            _settings = settings;
+            _settings.PropertyChanged += OnSettingsChanged;
```

Comment:
We don't know if they are getting loaded, LoadSettings basically just instantiates the page and then calls 'set' for every property.

---
In reply to: [81236099](https://github.com/Microsoft/RTVS/pull/2408#discussion_r81236099) [](ancestors = 81236099)